server/application/terminal.go

```diff
@@ -135,6 +128,18 @@ func (s *terminalHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	a, err := s.appLister.Get(app)
+	if err != nil {
+		http.Error(w, "Cannot get app", http.StatusBadRequest)
```

Comment:
We can't tell for sure that this is a bad request issue. We need to first check the returned error:
```Go
if err != nil {
    if apimachineryerrors.IsNotFound(err) {
        // don't log error
        // http 404
    } else {
        // log error
        // http 5xx
    }
}
```

Reply:
Agreed. 5xx is the behavior in every other case where we use `apiLister.Get`. Should we follow up with a general fix and announce the API behavior change, rather than having two different behaviors for (basically) the same error?

Reply:
Agree that we should have a general fix for this standardizing the code. However I'm not too sure about API breaking change as we are still returning 5xx when it is an internal error.

Reply:
Update: I'm very wrong here, should definitely check for the 404 and return a 5xx for other errors. Thanks Leo!  :-)